pyrefly/lib/commands/config_migration.rs

```diff
@@ -584,4 +578,105 @@ test = true
+
+        assert!(updated_content.contains("[project]"));
+        assert!(updated_content.contains("name = \"test-project\""));
+        assert!(updated_content.contains("[tool.poetry]"));
+        assert!(updated_content.contains("[tool.black]"));
+
+        assert!(updated_content.contains("[tool.pyrefly]"));
+        assert!(updated_content.contains("project_includes = [\"new/path/**/*.py\"]"));
+        assert!(!updated_content.contains("project_includes = [\"old/path/**/*.py\"]"));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_pyrefly_section_ordering() -> anyhow::Result<()> {
+        let tmp = tempfile::tempdir()?;
+        let new_pyproject_path = tmp.path().join("new_pyproject.toml");
+
+        let mut config = ConfigFile::default();
+        config.project_includes = Globs::new(vec!["test/**/*.py".to_owned()]);
+
+        let config_pyproject = PyProject::new(config);
+        let serialized = toml::to_string_pretty(&config_pyproject)?;
+        fs_anyhow::write(&new_pyproject_path, serialized.as_bytes())?;
+
+        let content = fs_anyhow::read_to_string(&new_pyproject_path)?;
+        println!("NEW FILE CONTENT:\n{}", content);
+        assert!(content.contains("[tool.pyrefly]"));
+
+        let existing_file_path = tmp.path().join("existing_file.toml");
+        fs_anyhow::write(&existing_file_path, b"[tool]\n# Some comment\n")?;
+
+        let mut config = ConfigFile::default();
+        config.project_includes = Globs::new(vec!["test/**/*.py".to_owned()]);
+        write_pyproject(&existing_file_path, config)?;
+
+        let content = fs_anyhow::read_to_string(&existing_file_path)?;
+        println!("EXISTING FILE CONTENT:\n{}", content);
+        assert!(content.contains("[tool.pyrefly]"));
+
+        let mypy_path = tmp.path().join("mypy_test.toml");
+        fs_anyhow::write(&mypy_path, b"[tool.mypy]\nfiles = [\"a.py\"]\n")?;
+        let mut config = ConfigFile::default();
+        config.project_includes = Globs::new(vec!["mypy_test.py".to_owned()]);
+        write_pyproject(&mypy_path, config)?;
+
+        let content = fs_anyhow::read_to_string(&mypy_path)?;
+        println!("MYPY TEST CONTENT:\n{}", content);
+        assert!(content.contains("[tool.pyrefly]"));
+
+        let ordering_path = tmp.path().join("ordering_test.toml");
+        let existing_content = r#"[project]
+name = "test-project"
+version = "0.1.0"
+
+# Comment before tool section
+[tool]
+# Comment within tool section
+
+[tool.a]
+option = "first"
+"#;
+        fs_anyhow::write(&ordering_path, existing_content.as_bytes())?;
+
+        let mut config = ConfigFile::default();
+        config.project_includes = Globs::new(vec!["ordering_test.py".to_owned()]);
+        write_pyproject(&ordering_path, config)?;
+
+        let content = fs_anyhow::read_to_string(&ordering_path)?;
+        println!("ORDERING TEST CONTENT:\n{}", content);
+        assert!(content.contains("[tool.pyrefly]"));
```

Comment:
Maybe for this test, to check ordering stuff, append the literal string of what you expect to be added to `existing_content` and compare that with the deserialized content here? It seems like that would check that ordering is working as expected.

Reply:
Updated in 7cb7ab03e04b3f21677044e7da7730d935f17055.